pkg/daemon/daemon.go

```diff
@@ -467,7 +467,7 @@ func (dn *Daemon) syncNode(key string) error {
 		}
 		if dn.isLayeredNode() {
 			// experimentalUpdateLayeredConfig is idempotent
-			return dn.experimentalUpdateLayeredConfig()
+			return (&ImageUpdater{&CoreOSDaemon{dn}}).experimentalUpdateLayeredConfig()
```

Comment:
Instead of having these everywhere, why not create a couple of constructors?

```golang
func newImageUpdater(cdn *CoreOSDaemon) *ImageUpdater {
	return &ImageUpdater{cdn}
}

func newMCUpdater(dn *Daemon) *MCUpdater {
	return &MCUpdater{dn}
}
```